### PR TITLE
cut down number of Tuple types a bit

### PR DESCRIPTION
**Pull request by JeffBezanson**
The system image contains a large number of Tuple types (~60000). This cuts down a bit by (1) removing unused types from the cache (~ 2000), (2) avoiding making copies of `Tuple` (there are only a couple hundred of them, but they are very easy to avoid so why not).

### CODE REVIEW

File: src/dump.c [hunk 1]
```diff
@@ -1916,7 +1930,11 @@ static void jl_save_system_image_to_stream(ios_t *f)
     jl_serialize_value(f, jl_typeinf_func);
     jl_serialize_value(f, jl_type_type->name->mt);
 
-    intptr_t i=2;
```
Comment (vtjnash) [inline src/dump.c]: need to restore the code to scan generated code for references to these (https://github.com/JuliaLang/julia/commit/735316ec6158ddd0716e3a3f839eb3dd343058e7#diff-669d4cc5c9c8f4573c5f8d57f5dcab20L336) before erasing them

**Comment by JeffBezanson** [comment]
@vtjnash How does this look?

**Comment by vtjnash** [comment]
apparently my review comment was lost in the rebase:

>need to restore the code to scan generated code for references to these (735316e#diff-669d4cc5c9c8f4573c5f8d57f5dcab20L336) before erasing them

**Comment by vtjnash** [comment]
oh, i see `prune_tcache` was updated too. yeah, lgtm

**Pull request merged**